### ISSUE TITLE
IP ranges do not round trip correctly as strings

### ISSUE THREAD

**Issue by hlascelles** [ip-num/ip-num#85]
We are seeing this:

```typescript
const cidr = "1.2.3.4/5";
IPv4CidrRange.fromCidr(cidr).toCidrString();

// -> 0.0.0.0/5
```

The CIDR we put in ("1.2.3.4/5") is not the CIDR we get out ("0.0.0.0/5")

I can see roughly why this is happening, as `/5` is a very big range. 

The problem for us is that `1.2.3.4/5` is used as an internal placeholder in CDK: https://github.com/aws/aws-cdk/blob/cea1039e3664fdfa89c6f00cdaeb1a0185a12678/packages/%40aws-cdk/aws-ec2/lib/vpc.ts#L2198

We are seeing tests break when we render our templates and try and round trip things. 

What would "you expect to happen" if you put string into a CIDR, then call `toCidrString` on it? I would say it should be the same, but I am sure there are some optimisations going on internally that are causing this.

**Comment by dadepo** [comment]
Thanks for reporting this. I'll take a look over the weekend and try to come up with a fix.

**Comment by dadepo** [comment]
Ok so took a quick look into this and here is why this is happening.

`IPv4CidrRange` creates a cidr representation with the actual start ip of the range. 

In the case of  `1.2.3.4/5` the start ip is `0.0.0.0` so when `toCidrString` is called, the cidr in string includes the start ip of the range and hence `0.0.0.0/5` is returned.

To further explain the current behaviour, Taking another example of `255.255.255.10/24` range which actually means, an IP range that starts at `255.255.255.0` with 256 ip addresses. 

`255.255.255.10/24` is strictly really not the range, but `255.255.255.10/24` is. And hence if you form a `IPv4CidrRange` and convert it back to cidr string, `255.255.255.0/24` will be returned.  This is what is going on with `1.2.3.4/5` also.

Question now is, should this current behaviour remain how it is, and user's of the library take this into consideration?

Right now, not sure what the answer should be. I'll give it some thoughts.

**Comment by dadepo** [comment]
@hlascelles I see this as a bug. And converting to `IPv4CidrRange` from string and back, should return the initial string. I'll update you when a new release is pushed

**Comment by hlascelles** [comment]
Terrific thank you. Thanks for this package, it's very useful!

**Comment by dadepo** [comment]
Should now be fixed in v1.5.0
